File: typescript/62/62.unique-paths.ts
/*
 * @lc app=leetcode id=62 lang=typescript
 *
 * [62] Unique Paths
 */

// @lc code=start
function uniquePaths(m: number, n: number): number {
    const grid = new Array(n).fill(1);

    for (let i = 1; i < m; i++) {
        for (let j = 1; j < n; j++) {
            grid[j] = grid[j] + grid[j - 1];
        }
    }

    return grid[n - 1]

};
// @lc code=end

File: rust/src/bin/62/62.unique-paths.rs
/*
 * @lc app=leetcode id=62 lang=rust
 *
 * [62] Unique Paths
 */

// @lc code=start
impl Solution {
    pub fn unique_paths(m: i32, n: i32) -> i32 {
        let mut grid = vec![1; n as usize];


        for i in 1..(m as usize) {
            for j in 1..(n as usize) {
                grid[j] = grid[j - 1] + grid[j];

            };
        }

        grid[(n - 1) as usize]
    }
}
// @lc code=end
